### ISSUE TITLE
Create a layer with identical neurons

### ISSUE THREAD

**Issue by wagenaartje** [cazala/synaptic#202]
I want to create a layer with identical neurons in a `Network`. Identical in this case means sharing the same bias and sharing the same weight. 

I'm not doing an ALL to ALL connection between layers, so please don't ask why I need identical neurons.

So the network could look like: 
input => Layer(50)
hidden1 => Layer(10)
output => Layer(4)

There are 10 neurons in the hidden layer, but each of those 10 neurons only gets input from 5 neurons from the input layer. 

How can I make the neurons share the same weights and biases, even after propagating?

**Comment by Jabher** [comment]
Theoretically - it's possible, but backprop is kind of hardcoded in synaptic, so it is impossible (at least I think so, maybe some dirty hack will work) to apply multiple anti-gradients to same value.

On other hand - possibly, some kind of reshape to make this work will do good things.

Also - can you please share the use case? I can try to help you to design the reshaping. 
Also we're working on ver.2 now, and analyzing the use-cases is very important to make API support all common cases.

**Comment by wagenaartje** [comment]
Well I'm trying to make a convolutional neural network. I have managed to create the structure of one by combining multiple perceptrons in a certain way, but this network cannot be backpropagated.

So I want to make one big network which has a layer(s) with same weights/biases to represent the layer that finds certain features in the input image. So every 5x5 input grid has to be connected with one neuron of this layer. 

But guess it is not possible, so I'm all hyped for ver. 2! I'm hoping there is some kind of implementation of a convolutional layer.
